zjit/src/hir.rs

```diff
@@ -3214,16 +3261,20 @@ pub fn iseq_to_hir(iseq: *const rb_iseq_t) -> Result<Function, ParseError> {
     let BytecodeInfo { jump_targets, has_blockiseq } = compute_bytecode_info(iseq);
     let mut insn_idx_to_block = HashMap::new();
     for insn_idx in jump_targets {
-        if insn_idx == 0 {
-            todo!("Separate entry block for param/self/...");
-        }
         insn_idx_to_block.insert(insn_idx, fun.new_block(insn_idx));
     }
 
     // Check if the EP is escaped for the ISEQ from the beginning. We give up
     // optimizing locals in that case because they're shared with other frames.
     let ep_escaped = iseq_escapes_ep(iseq);
 
+    // Compile an interpreter entry to the first block
+    let first_block = insn_idx_to_block[&0];
+    compile_entry_block(fun.entry_block, first_block, &mut fun, false);
```

Comment:
For later: let's get rid of the bool params

Reply:
Well, it's going to be so different between them that it probably doesn't make sense to share them anymore. I guess we'll just implement them in different functions while still sharing the entry_state logic among them.